### DISCUSSION TITLE
Advanced RLS via another table

### DISCUSSION THREAD

**Discussion by seabysalt** [supabase/supabase#18761] [Questions]
Hey everyone!

I'm currently struggling to implement a row level security policy and looking for pointers on how to go about this:

Here are the tables I'm working with:

- auth.users -- supabase auth table 
- public.profiles -- profiles table for app users
   ---> has a column "id" that is a reference to auth.users, saying id= auth.uuid() 
   ---> has a column "teamID" that is of type uuid
- public.teams -- teams table
   --> has a unique, primary key column called "id" that has a foreign key relation to the "teamID" column in public.profiles

Each user in profile can only belong to one team, saying the teamID and the id in public.profiles are unique.

```
auth.users:
| id  | ... |
|-----|-----|
| abc | ... |
| def | ... |
| ghi | ... |
```

```
public.profiles:
| id  | teamID        | ... |
|-----|---------------|-----|
| abc | example123    | ... | 
| def | example456    | ... |
| ghi | example789    | ... |
```

```
public.teams:
| id          | name      | ... |
|-------------|-----------|-----|
| example123  | teamName1 | ... |
| example456  | teamName2 | ... |
| example789  | teamName2 | ... |
```

Now, let's say I want a user to update the name in the public.teams table but I want to restrict it in a way that a user can only update the row that is specific to him/herself. Saying, in my update request, I send the id for the public.teams.id and then I want to check & only allow updating those rows, where this id that I sent is equal to the public.profiles.teams.teamsID where the public.profiles.id = auth.uid().

I tried different scenarios but nothing seems to be working. Maybe I cannot only update it with `.match({ id: teamID });`? Maybe someone could show me how to add the RLS correctly and how to phrase this update request? 

Any help would be greatly appreciated, thanks so much in advance 🙏

**Comment by GaryAustin1** [comment]
https://supabase.com/docs/guides/auth/row-level-security#using-joins
I think that is pretty close to what you want to do.  At a minimum it shows how to do the join on another table.
